### ISSUE TITLE
pip install tflite-runtime fails on Fedora 36, possibly due to Python 3.10

### ISSUE THREAD

**Issue by vorburger** [tensorflow/tensorflow#58264]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Build/Install

### Source

binary

### Tensorflow Version

2.10

### Custom Code

No

### OS Platform and Distribution

Linux Fedora 36

### Python version

3.10.7

### Current Behaviour?

```shell
$ cat /etc/os-release | head -2
NAME="Fedora Linux"
VERSION="36 (Workstation Edition)"

$ python --version
Python 3.10.7

$ python3 -m pip install tflite-runtime

ERROR: Could not find a version that satisfies the requirement tflite-runtime (from versions: none)
ERROR: No matching distribution found for tflite-runtime
```

NB that `python --version` on this Fedora Fedora 36 (Workstation) says `Python 3.10.7`...

... and a friend of mine pointed out that this could well be because on https://pypi.org/project/tflite-runtime/ we can see Python 3.7, 3.8, 3.9 - but not 3.10, yet. (Whereas e.g. https://pypi.org/project/numpy/ already has 3.10 and 3.11, and I CAN install that.)

Perhaps you would like to consider relaxing those version constraints, if that's possible? Or publish tflite for more recent Python versions already as well? Please let me know if this is easy enough to contribute a PR for an test - I'm happy to help! (Full disclosure: I'm a Googler  at work, playing with this on the weekend at home on a personal computer.)

In the mean time, I'll try to work around this with https://www.tensorflow.org/lite/guide/build_cmake_pip, or perhaps just use full Tensorflow instead of Lite and learn ;-) how to adapt project-keyword-spotter...

Background: I'm a new and noob user and wanted to figure out how to run e.g. https://github.com/google-coral/project-keyword-spotter/ on my workstation instead of on my Coral Dev board. https://www.tensorflow.org/lite/guide/python says that I should do `python3 -m pip install tflite-runtime` but this fails on Fedora 36 (Workstation), as described above.
```


### Standalone code to reproduce the issue

```shell
You would have to try on Fedora.

I could build a simple reproduce using a container, if that would of any use for CI of this project?
```

### Relevant log output

_No response_</details>

**Comment by vorburger** [comment]
BTW https://pypi.org/project/tensorflow/ has _Python :: 3.10_ - perhaps it would be sensible to publish both `tensorflow` and `tflite-runtime` for the same Python versions on PyPI? I'm happy to try to propose a PR doing that, if you want to point me to where this is specified.

PS: It would be neat if `pip install` would be more helpful here... see https://github.com/pypa/pip/issues/7565, https://github.com/pypa/pip/issues/8831, https://github.com/pypa/pip/issues/4228, https://github.com/pypa/pip/issues/6526.

> In the mean time, I'll try to work around this with https://www.tensorflow.org/lite/guide/build_cmake_pip

FTR: That seems to require more RAM than I have, and at least on a 20 GB laptop kicks in [the OOM killer](https://man.archlinux.org/man/systemd-oomd.service.8.en) 😿

**Comment by johnthagen** [comment]
This is essentially a duplicate of:

- https://github.com/tensorflow/tensorflow/issues/56137#issuecomment-1156723441

@terryheo mentioned he could do this:

- https://github.com/tensorflow/tensorflow/issues/56137#issuecomment-1156729053

But unfortunately it has been several months and the [latest `tflite-runtime-nightly`](https://pypi.org/project/tflite-runtime-nightly/2.12.0.dev20221023/#files) still only has Python 3.7-3.9 wheels.

**Comment by mihaimaruseac** [comment]
TFLite team and TF team have different release processes, although the code resides in the same repository.

**Comment by johnthagen** [comment]
@mihaimaruseac Would you happen to know where the build automation for `tflite-runtime` is if community members wanted to like to try to help update it to build 3.10 wheels?

**Comment by mihaimaruseac** [comment]
I'd triage this to @terryheo or someone else from TFLite team. I left TF in June and only do OSS drive-by work now

**Comment by sachinprasadhs** [comment]
You can close this issue and track the progress in the ongoing thread in the issue here https://github.com/tensorflow/tensorflow/issues/56137, , unless if you like to convert this to a feature request as mentioned below.
> BTW https://pypi.org/project/tensorflow/ has _Python :: 3.10_ - perhaps it would be sensible to publish both `tensorflow` and `tflite-runtime` for the same Python versions on PyPI? I'm happy to try to propose a PR doing that, if you want to point me to where this is specified.

**Comment by johnthagen** [comment]
Bump. This issue still needs resolved.

**Comment by bperseghetti** [comment]
@sachinprasadhs is there any update on this issue? Or is there another way to avoid this blocking users on with python 3.10 (arm64 Ubuntu 22.04)?

**Comment by johnthagen** [comment]
@bperseghetti 

See 

-https://github.com/tensorflow/tensorflow/issues/56137#issuecomment-1342705829

We're basically blocked on either DeadSnakes `armhf` builds being fixed, which some of us in the community tried to help with but we're unsuccessful, OR bumping the Docker image builds up to Ubuntu `20.04`.

As a community member, any help with these would be greatly appreciated as this is a big blocker for many of us.

**Comment by bperseghetti** [comment]
@johnthagen are you on the TF Discord? I just joined it and will post in bugs-and-issues. I did change the make file to build 3.10 through docker on 22.04 for arm64. I did it by changing the numpy dependency. How tightly coupled is TF to numpy version?

**Comment by bperseghetti** [comment]
If you are stuck in my boat here is a solution that I created: https://github.com/tensorflow/tensorflow/issues/56137#issuecomment-1367563496 might work for some of you.

**Comment by johnthagen** [comment]
Some notes on a summary of the blockers here

- https://github.com/tensorflow/tensorflow/issues/56137#issuecomment-1397351028

**Comment by johnthagen** [comment]
Should be fixed by 

- #59423

**Comment by pjpratik** [comment]
Hi @vorburger 

The tflite-runtime wheels for 3.10 have now been merged with the PR #59423 

https://github.com/tensorflow/tensorflow/blob/012530303cf66c15a7dc431d62e78e28a66858cc/tensorflow/lite/tools/pip_package/Makefile#L16-L18

https://github.com/tensorflow/tensorflow/blob/012530303cf66c15a7dc431d62e78e28a66858cc/tensorflow/lite/tools/pip_package/setup_with_binary.py#L53

Thanks.

**Comment by johnthagen** [comment]
@pjpratik I see that a new stable release of `tflite-runtime` was made `2.12.0`

- https://pypi.org/project/tflite-runtime/2.12.0/#files

But this one does not include the 3.10 wheels (`cp310`) that have been in `tflite-runtime-nightly` for a long time

- https://pypi.org/project/tflite-runtime-nightly/2.14.0.dev20230510/#files

Did something go wrong with the stable release?

CC @terryheo

**Comment by terryheo** [comment]
For the changes needed to support Python 3.10 wasn't included in r2.12 branch.
The Python 3.10  package will be available from r2.13 releases.
